src/graphql/root/mutation/user-update-username.ts

```diff
@@ -30,7 +31,9 @@ const UserUpdateUsernameMutation = GT.Field({
 
     if (result instanceof Error) {
       const appErr = mapError(result)
-      return { errors: [{ message: appErr.message || appErr.name }] }
+      return result instanceof UsernameIsImmutableError
+        ? { errors: [{ message: appErr.message || appErr.name }], user: domainUser }
+        : { errors: [{ message: appErr.message || appErr.name }] }
```

Comment:
this is not necessary `message: appErr.message || appErr.name` please use only `[mapError(result)]`

Reply:
I tested this, see: https://github.com/GaloyMoney/galoy/pull/1125#discussion_r813879550